### PR TITLE
Fix: disabling all logs by default.

### PR DESCRIPTION
**Pull request by carlosdnba**
In order to sort out issue #299, this commit disables all the logs by default.

They can still be turned on through the environment variable `ENABLE_WARMUP_LOGS`, like:
```
ENABLE_WARMUP_LOGS=true
```

This commit has passed all the unit tests.

### CODE REVIEW

**Comment by juanjoDiaz** [comment]
Hi @carlosdnba  thanks for the PR.

I think that we should use a setting instead of an env variable.
Also, I think that exceptions should always be logged.

**Comment by carlosdnba** [comment]
Hmm, that makes sense, let me put the exceptions logs back again.

**Comment by carlosdnba** [comment]
Hi @juanjoDiaz!

I have just pushed the two updates related to your suggestions.

File: src/warmer.js [hunk 1]
```diff
@@ -186,15 +190,15 @@ module.exports.warmUp = async (event, context) => {
 
     try {
       await Promise.all(Array(concurrency).fill(0).map(async () => await lambda.invoke(params).promise()));
-      console.log(\`Warm Up Invoke Success: \${func.name}\`);
+      logger(\`Warm Up Invoke Success: \${func.name}\`);
       return true;
     } catch (e) {
       console.log(\`Warm Up Invoke Error: \${func.name}\`, e);
```
Comment (juanjoDiaz) [inline src/warmer.js:196]: Please change this to `console.error(...)` to be more clear

File: src/warmer.js [hunk 2]
```diff
@@ -143,28 +143,32 @@ const lambda = new AWS.Lambda({
 });
 const functions = ${JSON.stringify(functions, null, '  ')};
 
+function logger(str) {
```
Comment (juanjoDiaz) [inline src/warmer.js:146]: This evaluate `enableLogs` for every time that logger is called.

```js
function logger(str) {
  ${enableLogs ? 'console.log(str);' : ''}
}
```

File: src/config.js [hunk 3]
```diff
@@ -41,6 +41,7 @@ function getWarmerConfig(config, defaultOpts) {
       ? config.environment
       : defaultOpts.environment,
     tracing: (config.tracing !== undefined) ? config.tracing : defaultOpts.tracing,
+    enableLogs: (config.enableLogs !== undefined) ? config.enableLogs : defaultOpts.enableLogs,
```
Comment (juanjoDiaz) [inline src/config.js:44]: I would rename this to `verbose`

File: src/config.js [hunk 4]
```diff
@@ -144,6 +145,7 @@ function getConfigsByWarmer(service, stage) {
     memorySize: 128,
     name: `${service.service}-${stage}-warmup-plugin-${warmerName}`,
     events: [{ schedule: 'rate(5 minutes)' }],
+    enableLogs: false,
```
Comment (juanjoDiaz) [inline src/config.js:148]: the default should be true to not break functionality

**Review by juanjoDiaz — CHANGES_REQUESTED** [review]
Thanks for the PR. It's looking pretty good.

I added a few small change requests.
Also, before this is merged, it need tests to maintain coverage and docs in the readme.

File: src/warmer.js (hunk 1 above)
Comment (carlosdnba) [reply src/warmer.js]: The `console.error` approach is resulting in a bunch of warnings when running the unit tests. Is that ok?

image

**Comment by carlosdnba** [comment]
@juanjoDiaz I have documented the new `verbose` option on the README.md. However, I am having difficulty setting unit tests to check inside the warmer lambda code if the console.log is or is not set. The coverage has not decreased a lot, though (it went from 99.21% to 97.72% on the branch column).

```sh
-----------|---------|----------|---------|---------|-------------------
File       | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
-----------|---------|----------|---------|---------|-------------------
All files  |     100 |    97.72 |     100 |     100 |
 config.js |     100 |    97.26 |     100 |     100 | 25,44
 index.js  |     100 |      100 |     100 |     100 |
 schema.js |     100 |      100 |     100 |     100 |
 utils.js  |     100 |      100 |     100 |     100 |
 warmer.js |     100 |    95.45 |     100 |     100 | 147
-----------|---------|----------|---------|---------|-------------------

Test Suites: 4 passed, 4 total
Tests:       88 passed, 88 total
```

**Comment by juanjoDiaz** [comment]
Sorry for the delay.
I'll merge this into a temporary branch and fix the tests and all that.

**Comment by carlosdnba** [comment]
> 

Thank you!

**Pull request merged**